lib/tag.js

```diff
@@ -71,7 +71,7 @@ Tag.createWithSignature = function(
   signingCallback
 ) {
   let tagBuffer;
-  return Tag.createBuffer(repo, tagName, target, tagger, message)
+  return Tag.createBuffer(repo, tagName, target.id(), tagger, message)
```

Comment:
Can we check if `target.id` exists, and if not throw a deprecation warning while still functioning? That way we give users a heads up that they're using the API incorrectly without abruptly breaking things for them. Something like (pseudocode, not tested)
```
const deprecatedCreateWithSignatureHelper = util.deprecate(function(oidTarget) {
  return repo.getCommit(oidTarget);
}, "Tag.createWithSignature target should be a Git Object, not Oid");

Tag.createWithSignature = async function(
  ...


  if (!target.id) {
   target = await deprecatedCreateWithSignatureHelper(target);
  }

...  
```

See https://nodejs.org/api/util.html#utildeprecatefn-msg-code if you aren't familiar with `util.deprecate`, we use this elsewhere in NodeGit.

Reply:
Done.